### ISSUE TITLE
Move geometric queries to apfGeometry + clean up implementation

### ISSUE THREAD

**Issue by bgranzow** [SCOREC/core#90]
The geometric queries defined in sam.cc:

* [withinBox](https://github.com/SCOREC/core/blob/master/sam/sam.cc#L80)
* [withinCyl](https://github.com/SCOREC/core/blob/master/sam/sam.cc#L89)

would make much more sense to be included in apfGeometry:

https://github.com/SCOREC/core/blob/master/apf/apfGeometry.h

---

Additionally, the logic for withinCyl looks broken on [this](https://github.com/SCOREC/core/blob/8c462070cc75172b931cef0ee08079762b01babe/sam/sam.cc#L102) line:

Essentially, the conditional is defined as:
```
if (fabs(a) <=b && sqrt( c <= d) )
```
which I'm not even sure makes sense. My initial guess is that the second to last parentheses on that line is misplaced, and the logic should look like:
```
if(fabs(a) <= b && sqrt(c) <= d)
```

---

Additionally, the implementation of these two geometry functions could be greatly improved by modifying the APIs to use APF objects, like
```
withinCyl(apf::Vector3 point, apf::Vector3 const& cyl_data_1, apf::Vector3 const& cyl_data_2, const double cyl_data_3, const double cyl_data_4);
```
where presumably the `cyl_data*` inputs have more meaningful names they could be given. Once this is done, the math in the `withinCyl` implementation can be greatly cleaned up using the overloaded operators that APF provides for `apf::Vector3`'s

**Comment by yangf4** [comment]
Thank you @bgranzow .  Let me fix this.

**Comment by yangf4** [comment]
fix is here: d506f51747b3810bb2dfd00cff38901d19296829
